drivers/rtc/rtc_ll_stm32.c

```diff
@@ -156,13 +156,15 @@ static int rtc_stm32_set_time(const struct device *dev, const struct rtc_time *t
 
 static int rtc_stm32_get_time(const struct device *dev, struct rtc_time *timeptr)
 {
-	ARG_UNUSED(dev);
+	const struct rtc_stm32_config *cfg = dev->config;
+	uint32_t sync_prescaler = cfg->ll_rtc_config.SynchPrescaler;
 
-	uint32_t rtc_date, rtc_time;
+	uint32_t rtc_date, rtc_time, rtc_subsecond;
 
-	/* Read time and date registers */
-	rtc_time = LL_RTC_TIME_Get(RTC);
-	rtc_date = LL_RTC_DATE_Get(RTC);
+	/* Read subsecond, time and date registers */
+	rtc_subsecond = LL_RTC_TIME_GetSubSecond(RTC);
+	rtc_time      = LL_RTC_TIME_Get(RTC);
+	rtc_date      = LL_RTC_DATE_Get(RTC);
```

Comment:
Reference manual says : "To ensure consistency between the 3 values, reading either RTC_SSR or RTC_TR locks the values in the higher-order calendar shadow registers until RTC_DR is read".
As I understand it reading RTC_SSR locks RTC_TR and RTC_DR thus allowing consistent reading. So there is no need to do the "do {} while();" thing that is done in #61580 imho if reading is done in the correct order.

Reply:
I didn't know that, I just remembered what was done in the other PR.
So I looked a bit more into it, and that's only valid if you use the shadow registers. But STM32F2x series doesn't have them (and F1x but that's a completely different RTC), and in the current code, LL_RTC_EnableShadowRegBypass is called in the init function anyway. So in any case, the shadow registers are not used, and the "do {} while();" is necessary to ensure consistency.
Even if you remove the call and actually use the shadow register, you will still need a special read for F2x.

Reply:
Oh waw ! Indeed Shadow register bypass is enabled… Don't know why… Ok so I will push a fix.

Reply:
It may have been enabled so RTC works correctly for any clock configuration. (One must ensure that APB clock is more than seven times the RTCCLK frequency to use shadow registers).
Requested fix is done in [01ce509](https://github.com/zephyrproject-rtos/zephyr/pull/62304/commits/01ce5094652f0be7acf4cd188779564ab76d3826).